nfsv4/nfs4_internal.h

```diff
@@ -0,0 +1,122 @@
+#include <nfs4.h>
+#include <stdio.h>
+#include <arpa/inet.h>
+#include <nfs4xdr.h>
+
+struct client {
+    int fd;
+    heap h;
+    u32 xid;
+    u32 address;
+    u64 clientid;
+    u8 session[NFS4_SESSIONID_SIZE];
+    u32 sequence;
+    u32 server_sequence;
+    u8 instance_verifier[NFS4_VERIFIER_SIZE];
+    boolean packet_trace;
+    buffer b;
+    bytes write_limit;
+    bytes read_limit;
+};
+
+
+struct file {
+    // filehandle
```

Comment:
It seems that we have `filehandle` already as the last field. Is `stateid` something that is to be added?

Reply:
yes to both